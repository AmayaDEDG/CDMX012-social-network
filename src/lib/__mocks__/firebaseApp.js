global.firebase = {
  initializeApp: () => {
  },
  auth: () => {
  },
//   firestore: () => {
//   },
//   // signInEmailAndPW: () => {
//   //   const signInWithEmailAndPassword = () => {};
//   // },
//   signInWithGoogle: () => {
//     const GoogleAuthProvider = () => {
//     };
//   },
//   signInWithFacebook: () => {
//     const FacebookAuthProvider = () => {
//     };
//   },
};
