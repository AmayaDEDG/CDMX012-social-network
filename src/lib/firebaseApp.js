/* eslint-disable arrow-body-style */
/* eslint-disable no-underscore-dangle */
/* eslint-disable max-len */
/* eslint-disable import/no-cycle */
/* eslint-disable no-param-reassign */
/* eslint-disable no-unused-vars */
//  Firebase implementation
const firebaseConfig = {
  apiKey: 'AIzaSyC26n4Fh-NfxC_ZNKZrFDH4NzrQrYwgirY',
  authDomain: 'petspace-3f65f.firebaseapp.com',
  projectId: 'petspace-3f65f',
  storageBucket: 'petspace-3f65f.appspot.com',
  messagingSenderId: '719999017536',
  appId: '1:719999017536:web:4e72654f1a5dba66b1b5a5',
};
firebase.initializeApp(firebaseConfig);
// firebase.firestore().settings({ experimentalForceLongPolling: true, merge: true });

export const getCurrentUserName = () => {
  const user = firebase.auth().currentUser;
  const userName = user.displayName;
  return userName;
};
