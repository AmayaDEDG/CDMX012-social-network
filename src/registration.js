const registration = `
    <section class="reg-box">
        <p>Crea un usuario y contraseña</p>
        <form id="registrationForm" action="submit">
            <input placeholder="Nombre" id="regName"></input>
            <input placeholder="Correo electrónico" id="regEmail"></input>
            <input type="password" placeholder="Contraseña" id="regPW"></input>
            <input type="password" placeholder="Confirma contraseña" id="regPW2"></input>
            <button class="register-button" type="submit" id="regBtn">Regístrate</button>
        </form>
        <p>o regístrate con:</p>
        <img src="img/facebook.png" class="facebook-logo">
        <img src="img/google.png" class="gmail-logo">
    </section>
    <a href="#" onclick="onNavigate('/'); return false;"><button class="register-button">Regresa al inicio</button></a>
`