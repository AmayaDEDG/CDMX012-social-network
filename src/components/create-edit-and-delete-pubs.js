/* eslint-disable import/no-unresolved */
/* eslint-disable import/no-cycle */
import { getCurrentUserName } from '../lib/firebaseApp.js';
import { deletePost, addPostCollection, editPost } from '../lib/addFirestore.js';
import { accessPosts } from '../lib/readFirestore.js';

function modifyText(postId, postArea, userId) {
  const pubText = document.getElementById(`text${postId}`).innerText;
  console.log(pubText);
  const textSpace = document.getElementById(`textSapce${postId}`);
  textSpace.innerHTML = `
  <input class='to-modify-text' id='modifyText${postId}' value='${pubText}' />
  <img class='modify-text-icon' id='modifyTextIcon' src='img/modifyIcon.png' />
  `;
  const textInput = document.getElementById(`modifyText${postId}`);
  const end = textInput.value.length;
  textInput.setSelectionRange(end, end);
  textInput.focus();
  const modifyIcon = document.getElementById('modifyTextIcon');
  modifyIcon.addEventListener('click', () => {
    const modifiedText = document.getElementById(`modifyText${postId}`).value;
    textSpace.innerHTML = `<p class='publication-text' id='text${postId}'></p>`;
    const finalText = document.getElementById(`text${postId}`);
    finalText.innerHTML = modifiedText;
    editPost(userId, postId, postArea, modifiedText);
  });
}

export const editPublication = (user, postId, postArea, input) => {
  if (user === getCurrentUserName()) {
    const textOptions = document.getElementById(`optionSection${postId}`);
    textOptions.style.visibility = 'visible';
    const editInput = document.createElement('input');
    editInput.setAttribute('class', 'edit-input');
    console.log(postId);
    modifyText(postId, postArea, user, input);
  } else {
    const textOptions = document.getElementById(`optionSection${postId}`);
    textOptions.innerHTML = '';
  }
};

export const deletePublication = (user, postUser, postId, postArea) => {
  let publicationDeletedMsg = '';
  if (user === postUser) {
    if (window.confirm('¿Segurx que deseas borrar?')) {
      deletePost(postId, postArea);
      publicationDeletedMsg = 'esta publicación ha sido eliminada';
      return publicationDeletedMsg;
    }
  } publicationDeletedMsg = 'no se pudo eliminar';
  return publicationDeletedMsg;
};

export const publishPublication = (publicationInput, postArea) => {
  if (publicationInput === null || publicationInput === '') {
    alert('No has escrito una publicación');
  } else {
    addPostCollection(publicationInput);
    accessPosts(postArea);
  }
};
