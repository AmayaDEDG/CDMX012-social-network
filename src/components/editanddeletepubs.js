/* eslint-disable import/no-unresolved */
/* eslint-disable import/no-cycle */
import { getCurrentUserName } from '../lib/firebaseApp.js';
import { deletePost } from '../lib/firestore.js';

export const editPublication = (user, postId) => {
  if (user === getCurrentUserName()) {
    const editInput = document.createElement('input');
    editInput.setAttribute('class', 'edit-input');
    console.log(postId);
    // vincular con editPost(postId) en firebase, maybe un eventListener;
    console.log('hasllegadoaqui');
  } else {
    alert('no puedes editar');
  }
};

export const deletePublication = (user, postId) => {
  let publicationDeletedMsg = '';
  if (user === getCurrentUserName()) {
    deletePost(postId);
    publicationDeletedMsg = 'esta publicaci√≥n ha sido eliminada';
  } else {
    publicationDeletedMsg = 'no se pudo eliminar';
  }
  return publicationDeletedMsg;
};
