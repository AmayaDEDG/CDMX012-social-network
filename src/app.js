/* eslint-disable import/no-cycle */
//  Routing
import { home } from './home.js';
import { registration } from './registration.js';

//  Firebase implementation
const firebaseConfig = {
  apiKey: 'AIzaSyC26n4Fh-NfxC_ZNKZrFDH4NzrQrYwgirY',
  authDomain: 'petspace-3f65f.firebaseapp.com',
  projectId: 'petspace-3f65f',
  storageBucket: 'petspace-3f65f.appspot.com',
  messagingSenderId: '719999017536',
  appId: '1:719999017536:web:4e72654f1a5dba66b1b5a5',
};
export const firebaseInitialization = firebase.initializeApp(firebaseConfig);

export const errorTranslate = {
  'auth/invalid-email': 'El email es inválido.',
  'auth/email-already-in-use': 'El email ya está registrado.',
  'auth/weak-password': 'La contraseña es inválida',
  'auth/wrong-password': 'La contraseña es incorrecta',
  'auth/user-not-found': 'El usuario no existe',
};

//  Routing
const routes = {
  '/': home,
  '/registration': registration,
};

const root = document.getElementById('root');
root.appendChild(routes[window.location.pathname]());

export const onNavigate = (pathname) => {
  window.history.pushState(
    {},
    pathname,
    window.location.origin + pathname,
  );
  if (root.hasChildNodes()) {
    root.innerHTML = '';
  }
  root.appendChild(routes[pathname]());
};

window.onpopstate = () => {
  root.appendChild(routes[window.location.pathname]());
};
