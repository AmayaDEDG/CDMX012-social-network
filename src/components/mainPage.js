/* eslint-disable no-multiple-empty-lines */
/* eslint-disable import/no-cycle */
import { onNavigate } from '../app.js';

export const mainPage = () => {
  const mainPageStructureAttributes = {
    class: 'main-page',
    id: 'mainPage',
  };
  const menuUpAttributes = {
    class: 'menu-up menu',
    id: 'menuUp',
  };
  const petspacePlanetAttributes = {
    class: 'petspace-planet',
    id: 'petspacePlanet',
    src: 'img/PetSpacePlanet.png',
  };
  const shopIconAttributes = {
    class: 'shop-icon icon',
    id: 'shopIcon',
    src: 'img/shopIcon.png',
  };
  const locationIconAttributes = {
    class: 'location-icon icon',
    id: 'locationIcon',
    src: 'img/locationIcon.png',
  };
  const searchIconAttributes = {
    class: 'search-icon icon',
    id: 'searchIcon',
    src: 'img/searchIcon.png',
  };
  const homePageAttributes = {
    class: 'home-page',
    id: 'homePage',
  };
  const homePageMAttributes = {
    class: 'home-page-M',
    id: 'homePageM',
  };
  const menuDownAttributes = {
    class: 'menu-down menu',
    id: 'menuDown',
  };
  const homeIconAttributes = {
    class: 'home-icon icon',
    id: 'homeIcon',
    src: 'img/homeIcon.png',
  };
  const menuIconAttributes = {
    class: 'menu-icon icon',
    id: 'menuIcon',
    src: 'img/menuIcon.png',
  };
  const notificationIconAttributes = {
    class: 'notification-icon icon',
    id: 'notificationIcon',
    src: 'img/notificationIcon.png',
  };
  const messageIconAttributes = {
    class: 'message-icon icon',
    id: 'messageIcon',
    src: 'img/messageIcon.png',
  };
  const profileIconAttributes = {
    class: 'profile-icon icon',
    id: 'profileIcon',
    src: 'img/profileIcon.png',
  };

  const setAttributes = (element, attributes) => {
    Object.keys(attributes).forEach((attr) => element.setAttribute(attr, attributes[attr]));
  };

  const mainPageStructure = document.createElement('div');
  const menuUp = document.createElement('section');
  const petspacePlanet = document.createElement('img');
  const shopIcon = document.createElement('img');
  const locationIcon = document.createElement('img');
  const searchIcon = document.createElement('img');
  const homePage = document.createElement('section');
  const homePageM = document.createElement('p');
  const menuDown = document.createElement('section');
  const homeIcon = document.createElement('img');
  const menuIcon = document.createElement('img');
  const notificationIcon = document.createElement('img');
  const messageIcon = document.createElement('img');
  const profileIcon = document.createElement('img');
  // const addBr = document.createElement('br');

  setAttributes(mainPageStructure, mainPageStructureAttributes);
  setAttributes(menuUp, menuUpAttributes);
  setAttributes(petspacePlanet, petspacePlanetAttributes);
  setAttributes(shopIcon, shopIconAttributes);
  setAttributes(locationIcon, locationIconAttributes);
  setAttributes(searchIcon, searchIconAttributes);
  setAttributes(homePage, homePageAttributes);
  setAttributes(homePageM, homePageMAttributes);
  setAttributes(menuDown, menuDownAttributes);
  setAttributes(homeIcon, homeIconAttributes);
  setAttributes(menuIcon, menuIconAttributes);
  setAttributes(notificationIcon, notificationIconAttributes);
  setAttributes(messageIcon, messageIconAttributes);
  setAttributes(profileIcon, profileIconAttributes);

  menuUp.append(petspacePlanet, shopIcon, locationIcon, searchIcon);
  homePage.appendChild(homePageM);
  menuDown.append(homeIcon, menuIcon, notificationIcon, messageIcon, profileIcon);
  mainPageStructure.append(menuUp, homePage, menuDown);
  menuIcon.addEventListener('click', () => { onNavigate('/'); });

  return mainPageStructure;
};
