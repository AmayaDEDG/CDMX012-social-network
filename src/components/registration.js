/* eslint-disable max-len */
/* eslint-disable import/no-cycle */
import { onNavigate } from '../app.js';
import { createAccount, signInWithFacebook, signInWithGoogle } from '../lib/firebaseApp.js';

export const registration = () => {
  const registrationStructureAttributes = {
    class: 'registration',
  };
  const petspaceLogoAttributes = {
    class: 'petspace-logo',
    id: 'petspaceLogo',
    src: 'img/PetSpaceLogo.png',
  };
  const formAttributes = {
    id: 'registrationForm',
    action: 'submit',
  };
  const inputNameAttributes = {
    placeholder: 'Nombre',
    id: 'regName',
  };
  const inputEmailAttributes = {
    placeholder: 'Correo electrónico',
    id: 'regEmail',
  };
  const inputPWAttributes = {
    type: 'password',
    placeholder: 'Contraseña',
    id: 'regPW',
  };
  const messageAttributes = {
    id: 'regMessage',
  };
  const buttonRegisterAttributes = {
    class: 'register-button',
    type: 'button',
    id: 'regBtn',
  };
  const facebookLogoAttributes = {
    src: 'img/facebook.png',
    class: 'facebook-logo',
  };
  const gmailLogoAttributes = {
    src: 'img/google.png',
    class: 'gmail-logo',
  };
  const returnButtonAttributes = {
    class: 'return-button',
    id: 'returnButton',
  };

  const setAttributes = (element, attributes) => {
    Object.keys(attributes).forEach((attr) => element.setAttribute(attr, attributes[attr]));
  };

  const registrationStructure = document.createElement('div');
  const petspaceLogo = document.createElement('img');
  const registrationBox = document.createElement('section');
  registrationBox.setAttribute('class', 'reg-box');
  const pUser = document.createElement('p');
  const form = document.createElement('form');
  const inputName = document.createElement('input');
  const inputEmail = document.createElement('input');
  const inputPW = document.createElement('input');
  const regMessage = document.createElement('p');
  const eyeButton = document.createElement('button');
  eyeButton.setAttribute('id', 'seePasswordReg');
  const eye = document.createElement('i');
  eye.setAttribute('class', 'fas fa-eye');
  const buttonRegister = document.createElement('button');
  const pOptions = document.createElement('p');
  const facebookLogo = document.createElement('img');
  const gmailLogo = document.createElement('img');
  const returnButton = document.createElement('button');
  const addBr1 = document.createElement('br');
  const addBr2 = document.createElement('br');
  const addBr3 = document.createElement('br');
  const addBr4 = document.createElement('br');

  setAttributes(registrationStructure, registrationStructureAttributes);
  setAttributes(petspaceLogo, petspaceLogoAttributes);
  setAttributes(form, formAttributes);
  setAttributes(inputName, inputNameAttributes);
  setAttributes(inputEmail, inputEmailAttributes);
  setAttributes(inputPW, inputPWAttributes);
  setAttributes(regMessage, messageAttributes);
  setAttributes(buttonRegister, buttonRegisterAttributes);
  setAttributes(facebookLogo, facebookLogoAttributes);
  setAttributes(gmailLogo, gmailLogoAttributes);
  setAttributes(returnButton, returnButtonAttributes);

  pUser.textContent = 'Crea un usuario y contraseña';
  buttonRegister.textContent = 'Regístrate';
  pOptions.textContent = 'o regístrate con:';
  returnButton.textContent = 'Regresa al inicio';

  eyeButton.append(eye);
  form.append(inputName, addBr1, inputEmail, addBr2, inputPW, addBr3, regMessage, eyeButton, buttonRegister);
  registrationBox.append(pUser, form, pOptions, facebookLogo, gmailLogo);
  registrationStructure.append(petspaceLogo, registrationBox, addBr4, returnButton);

  returnButton.addEventListener('click', () => { onNavigate('/'); });
  eyeButton.addEventListener('click', (e) => {
    e.preventDefault();
    if (inputPW.type === 'password') {
      inputPW.type = 'text';
    } else {
      inputPW.type = 'password';
    }
  });

  buttonRegister.addEventListener('click', (e) => {
    e.preventDefault();
    const email = inputEmail.value;
    const password = inputPW.value;
    const message = regMessage;
    console.log(email, password);
    createAccount(message, email, password);
  });
  /*   firebase.auth().onAuthStateChanged((user) => {
    if (user) {
      console.log(user);
      // eslint-disable-next-line no-empty
    } else {
    }
  }); */

  gmailLogo.addEventListener('click', () => {
    signInWithGoogle();
  });
  facebookLogo.addEventListener('click', () => {
    signInWithFacebook();
  });
  return registrationStructure;
};
