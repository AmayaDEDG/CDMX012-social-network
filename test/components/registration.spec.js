/**
 * @jest-environment jsdom
 */
import '../../src/lib/__mocks__/firebaseApp';
import '../app';
import { registration } from '../../src/components/registration';

describe('registration', () => {
  it('Snapshot registration', () => {
    const component = registration();
    document.body.innerHTML = '<div id="root"></div>';
    const rootDiv = document.getElementById('root');
    rootDiv.appendChild(component);
    expect(rootDiv.innerHTML).toMatchSnapshot('./registration.spec.js.snap');
  });
  /* it('onNavigate funciona correctamente', () => {
    expect(window.location.pathname).toBe('/registration');
  }); */
  it('Ã­s a function', () => {
    expect(typeof registration).toBe('function');
  });
});
