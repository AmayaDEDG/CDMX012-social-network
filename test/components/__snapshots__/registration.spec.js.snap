// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home Snapshot home 1`] = `"<div><img class=\\"desktop-img\\" src=\\"img/homeDesktop.png\\"><div class=\\"structure\\"><img class=\\"petspace-logo\\" id=\\"petspaceLogo\\" src=\\"img/PetSpaceLogo.png\\"><section class=\\"reg-box\\"><p>Crea un usuario y contraseña</p><form id=\\"registrationForm\\" action=\\"submit\\"><input placeholder=\\"Nombre\\" id=\\"regName\\"><input placeholder=\\"Correo electrónico\\" id=\\"regEmail\\"><input type=\\"password\\" placeholder=\\"Contraseña\\" id=\\"regPW\\"><p id=\\"regMessage\\"></p><button id=\\"seePasswordReg\\"><i class=\\"fas fa-eye\\"></i></button><button class=\\"register-button\\" type=\\"button\\" id=\\"regBtn\\">Regístrate</button></form><p>o regístrate con:</p><img src=\\"img/facebook.png\\" class=\\"facebook-logo\\"><img src=\\"img/google.png\\" class=\\"gmail-logo\\"></section><button class=\\"return-button\\" id=\\"returnButton\\">Regresa al inicio</button></div></div>"`;
